packages/flutter/lib/src/rendering/object.dart

```diff
@@ -4583,133 +4328,609 @@ mixin RelayoutWhenSystemFontsChangeMixin on RenderObject {
+}
+
+/// A container used when a [RenderObject] wants to add multiple independent
+/// [_InterestingSemanticsFragment] to its parent.
+///
+/// The [_InterestingSemanticsFragment] to be added to the parent can be
+/// obtained via [mergeUpFragmentProviders].
+class _ContainerSemanticsFragment extends _SemanticsFragment {
+  _ContainerSemanticsFragment({
+    required super.dropsSemanticsOfPreviousSiblings,
+    required this.siblingMergeGroups,
+  });
+
+  final List<List<_InterestingSemanticsFragmentProvider>> siblingMergeGroups;
+
+  @override
+  void addAll(Iterable<_InterestingSemanticsFragmentProvider> provider) {
+    mergeUpFragmentProviders.addAll(provider);
+  }
+
+  @override
+  final List<_InterestingSemanticsFragmentProvider> mergeUpFragmentProviders = <_InterestingSemanticsFragmentProvider>[];
+}
+
+/// A [_SemanticsFragment] that describes which concrete semantic information
+/// a [RenderObject] wants to add to the [SemanticsNode] of its parent.
+///
+/// Specifically, it describes which children (as returned by [compileSemanticsNodes])
+/// should be added to the parent's [SemanticsNode] and which [config] should be
+/// merged into the parent's [SemanticsNode].
+abstract class _InterestingSemanticsFragment extends _SemanticsFragment {
+  _InterestingSemanticsFragment({
+    required _RenderObjectSemantics owner,
+    required super.dropsSemanticsOfPreviousSiblings,
+  }) : _ancestorsUntilParent = <_RenderObjectSemantics>[owner];
+
+  /// The [_RenderObjectSemantics] that owns this fragment.
+  _RenderObjectSemantics get owner => _ancestorsUntilParent.first;
+
+  final List<_RenderObjectSemantics> _ancestorsUntilParent;
+
+  List<_RenderObjectSemantics> get _ancestorChainUntilExplicitParent {
+    Iterable<_RenderObjectSemantics> result = _ancestorsUntilParent;
+    _InterestingSemanticsFragment? implicitParent = _implicitParent;
+    while (implicitParent != null) {
+      result = result.followedBy(implicitParent._ancestorsUntilParent.skip(1));
+      implicitParent = implicitParent._implicitParent;
+    }
+    return result.toList();
+  }
+
+  bool _hasAncestorWithExplicitChild = false;
+
+  /// The reference to the fragment that last ancestor in
+  /// [_ancestorsUntilParent] produced.
+  ///
+  /// Null if the last ancestor produce explicit _SwitchableFragment.
+  ///
+  /// This is needed to construct the complete render object ancestors chain
+  /// to the parent semantics node.
+  _InterestingSemanticsFragment? _implicitParent;
+
+  bool _shouldDrop(SemanticsNode node) => node.isInvisible;
+
+  /// Copies parent inherited data from another _InterestingSemanticsFragment.
+  void copyParentInheritedDataFrom(_InterestingSemanticsFragment other) {
+    other._ancestorsUntilParent.skip(1).forEach(addAncestor);
+    if (other._tagsForChildren?.isNotEmpty ?? false) {
+      addTags(other._tagsForChildren!);
+    }
+    _implicitParent = other._implicitParent;
+  }
+
+  /// The children to be added to the parent.
```

Comment:
Does this still only compile the children? It used to be called compileChildren, where this comment made more sense.